src/parse/asp/interpreter.go

```diff
@@ -19,35 +20,43 @@ type interpreter struct {
 	parser      *Parser
 	subincludes *cmap.Map[string, pyDict]
 
-	configs      map[*core.BuildState]*pyConfig
-	configsMutex sync.RWMutex
+	configs       map[*core.BuildState]*pyConfig
+	configsMutex  sync.RWMutex
+	pluginConfigs *cmap.Map[pluginConfigKey, *pyConfigBase]
 
 	breakpointMutex sync.Mutex
 	limiter         semaphore
 
 	stringMethods, dictMethods, configMethods map[string]*pyFunc
 }
 
+type pluginConfigKey struct {
+	Name  string
+	State *core.BuildState
+}
+
 // newInterpreter creates and returns a new interpreter instance.
 // It loads all the builtin rules at this point.
 func newInterpreter(state *core.BuildState, p *Parser) *interpreter {
 	s := &scope{
 		state:  state,
 		locals: map[string]pyObject{},
 	}
+	i := &interpreter{
+		scope:   s,
+		parser:  p,
+		configs: map[*core.BuildState]*pyConfig{},
+		limiter: make(semaphore, state.Config.Parse.NumThreads),
+	}
 	// If we're creating an interpreter for a subrepo, we should share the subinclude cache.
-	var subincludes *cmap.Map[string, pyDict]
 	if p.interpreter != nil {
-		subincludes = p.interpreter.subincludes
+		i.subincludes = p.interpreter.subincludes
+		i.pluginConfigs = p.interpreter.pluginConfigs
 	} else {
-		subincludes = cmap.New[string, pyDict](cmap.SmallShardCount, cmap.XXHash)
-	}
-	i := &interpreter{
-		scope:       s,
-		parser:      p,
-		subincludes: subincludes,
-		configs:     map[*core.BuildState]*pyConfig{},
-		limiter:     make(semaphore, state.Config.Parse.NumThreads),
+		i.subincludes = cmap.New[string, pyDict](cmap.SmallShardCount, cmap.XXHash)
+		i.pluginConfigs = cmap.New[pluginConfigKey, *pyConfigBase](cmap.SmallShardCount, func(key pluginConfigKey) uint64 {
+			return cmap.XXHash(key.Name) ^ uint64(uintptr(unsafe.Pointer(key.State)))
+		})
```

Comment:
Hmmm I feel like go must know how to hash these objects otherwise we wouldn't be able to use the config key in a map. Bit annoying that it doesn't expose that for us here.

Reply:
It has some way of doing it - I thought the stdlib map hash was exposed somewhere but I can't find it right now.
All I really need is the pointer value anyway...